CMSgov/easi-app
# [EASI-2948] Advice letter form save and exit

# EASI-2948

## Changes and Description

- Created `useEasiForm` hook that extends React Hook Form's `useForm` to add a `partialSubmit` function
  - `partialSubmit` clears error messages and updates only valid changed fields before executing callback
- Implemented `partialSubmit` on Advice Letter Form's "Save and Exit" button
- Removed submit handler from "back" buttons
- Fixed bug in step validation (form was redirecting user when navigating to Summary step from further in form)



## How to test this change

- Test advice letter form
  - Confirm that valid fields are updated and errors do not show when clicking "back" or "save and exit" buttons
  - Confirm that step navigation works as expected



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
